File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int start = 0;
        int end = numbers.size() - 1;

        while (start < end) {
            int sum_br = numbers[start] + numbers[end];

            if (sum_br > target) end--;
            else if (sum_br < target) start++;
            else return {start + 1, end + 1};
        }
        return {};
        
    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        for m in matrix:
            l = 0
            r = len(matrix[0]) - 1

            while l <= r:
                i = (r + l) // 2
                if m[i] > target:
                   r = i - 1
                elif m[i] < target:
                    l = i + 1
                else:
                    return True 

        return False